tests/main.test.js

```diff
@@ -56,5 +57,54 @@ describe("Verify Getters", () => {
             expectKeys(json_data, return_keys);
             expect(Object.keys(json_data["data"]).length).toBe(max_datapoints);
         });
+
+        test(`/${endpoint} Empty page load`, async() => {
+            const all_data = await cosi_request()
+                                    .get(`/${endpoint}`)
+                                    .expect(200)
+                                    .query({page: Number.MAX_SAFE_INTEGER})
+                                    .expect("Content-Type", /json/);
+            let json_data = JSON.parse(all_data.text);
+            expect(Object.keys(json_data["data"]).length).toBe(0);
+        });
+
+        test(`/${endpoint} Invalid page load`, async() => {
+            const fake_pages = ["cosi", "-1", "!@#$%^&*()-_+=`"];
+            for (const page of fake_pages){
+                const all_data = await cosi_request()
+                                    .get(`/${endpoint}`)
+                                    .query({page: `${page}`})
+                                    .expect(200)
+                                    .expect("Content-Type", /json/);
+                let json_data = JSON.parse(all_data.text);
+                expect(json_data["page"]).toBe(0);
+                expect(Object.keys(json_data["data"]).length).toBe(max_datapoints);
+                expect(json_data["total_pages"]).toBe(Math.ceil(total_datapoints_per_table/max_datapoints));
+            }
+        });
+
+        test(`/${endpoint} Correct page count`, async() => {
+            const all_data = await cosi_request()
+                                    .get(`/${endpoint}`)
+                                    .expect(200)
+                                    .expect("Content-Type", /json/);
+            let json_data = JSON.parse(all_data.text);
+            let total_pages = json_data["total_pages"];
+            expect(total_pages).toBe(Math.ceil(total_datapoints_per_table/max_datapoints));
+
+        });
+
+        test(`/${endpoint} load < 100ms`, async() => {
+            // Assert that all tables can load data page in < 100ms
+            let startTime = Date.now();
+            const response = await cosi_request()
+                                    .get(`/${endpoint}`)
+                                    .query({page: 0})
+                                    .expect(200)
```

Comment:
For perf tests, do the asserts need to be done after?

Reply:
I gave it a shot tonight, but I struggled to get a proper assertion function in this chain, so I just excluded the usual expects here with the justification that if there are issues with those, they'll be caught in all the other tests.

Reply:
Fair, got it.